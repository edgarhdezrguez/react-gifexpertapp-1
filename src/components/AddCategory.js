import React, {useState} from 'react'
import PropTypes from 'prop-types'

const AddCategory = ({setCategories}) => {
    // const [categories, setCategories] = catgs;
    const [inputValue, setInputValue] = useState('');
    const handleInputChange = (e) => {
        setInputValue(e.target.value);
        // console.log('handleInputChange call');
    };

    const handleSubmit = (e) =>{
        e.preventDefault();
        
        // console.log('handleSubmit called', inputValue);

        if (inputValue.trim().length > 0){
            setCategories(categories =>  [inputValue, ...categories]);
            setInputValue('');
            // console.log('in if');
        } else {
            //alert("Value dont be empty");
            console.log('in else');
        }
    };

    return (
        <form onSubmit={handleSubmit}>
            <p>{ inputValue }</p>
            <input 
                type="text" 
                value={inputValue}
                onChange={handleInputChange}
                name="addSerie" 
                id="serie2Add"/>
            {/* <button onClick={handleAdd}> Add </button> */}
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}

export default AddCategory;
