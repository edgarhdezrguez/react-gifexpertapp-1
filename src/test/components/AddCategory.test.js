import React from 'react'
import { shallow, configure } from 'enzyme';
import '@testing-library/jest-dom'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import AddCategory from '../../components/AddCategory';

configure({ adapter: new Adapter() });
describe('Testing AddCategory Component', () => {
    
    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories}/>),
        input,
        form;

    beforeEach(()=>{
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories}/>);
        input = wrapper.find('input');
        form = wrapper.find('form');
    });

    test('should be show correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should be change the textbox', () => {
        const value = 'Hi World';

        input.simulate('change', { target: { value } });

        //console.log(wrapper.find('p').text().trim());
        expect(wrapper.find('p').text().trim()).toBe(value);
    });

    test('should not be posted the information with submit', () => {
        
        form.simulate('submit', { preventDefault(){} });

        expect(setCategories).not.toHaveBeenCalled();

    });
    

    test('should be call setCategories funct and clear the textbox value', () => {
       
        const value = 'It called';
        input = wrapper.find('input').simulate('change', { target: { value } });
        form = wrapper.find('form').simulate('submit', { preventDefault(){} });

        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledWith( expect.any(Function) );
        expect(input.prop('value').trim()).toBe('');
    });
});
