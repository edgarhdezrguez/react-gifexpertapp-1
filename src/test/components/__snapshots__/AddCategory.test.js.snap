// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing AddCategory Component should be show correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    id="serie2Add"
    name="addSerie"
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
